### PR TITLE
refactor(trajectory_follower_nodes): use max_steer_angle in common

### PR DESCRIPTION
**Pull request by takayuki5168**
Signed-off-by: Takayuki Murooka <takayuki5168@gmail.com>

## Description

use max_steer_angle defined in vehicle_info_utils in mpc_follower

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/trajectory_follower/src/mpc_lateral_controller.cpp [hunk 1]
```diff
@@ -79,13 +79,13 @@ MpcLateralController::MpcLateralController(rclcpp::Node & node) : node_{&node}
   m_new_traj_end_dist = node_->declare_parameter<float64_t>("new_traj_end_dist");            // [m]
 
   /* mpc parameters */
-  const float64_t steer_lim_deg = node_->declare_parameter<float64_t>("steer_lim_deg");
+  const auto vehicle_info = vehicle_info_util::VehicleInfoUtil(*node_).getVehicleInfo();
+  const float64_t wheelbase = vehicle_info.wheel_base_m;
+  const float64_t steer_lim_deg = vehicle_info.max_steer_angle_rad;
```
Comment (TakaHoribe) [inline control/trajectory_follower/src/mpc_lateral_controller.cpp:84]: need unit conversion: rad to deg.

You can get rid of the `steer_lim_deg` and substitute the arg directly.
```
m_mpc.m_steer_lim = vehicle_info.max_steer_angle_rad;
```

**Review by taikitanaka3 — APPROVED** [review]
I confirm parameter is correctly set in mpc with max steer angle 0.0 for testing

ros2 param dump --print /control/trajectory_follower/controller_node_exe
/control/trajectory_follower/controller_node_exe:
  ros__parameters:
    admissible_position_error: 5.0
    admissible_yaw_error_rad: 1.57
    brake_keeping_acc: -0.2
    converged_steer_rad: 0.1
    ctrl_period: 0.03
    current_vel_threshold_pid_integration: 0.5
    curvature_smoothing_num_ref_steer: 15
    curvature_smoothing_num_traj: 15
    delay_compensation_time: 0.17
    drive_state_offset_stop_dist: 1.0
    drive_state_stop_dist: 0.5
    emergency_acc: -5.0
    emergency_jerk: -3.0
    emergency_state_overshoot_stop_dist: 1.5
    emergency_state_traj_rot_dev: 0.7
    emergency_state_traj_trans_dev: 3.0
    emergency_vel: 0.0
    enable_brake_keeping_before_stop: false
    enable_keep_stopped_until_steer_convergence: false
    enable_large_tracking_error_emergency: true
    enable_overshoot_emergency: true
    enable_path_smoothing: false
    enable_slope_compensation: false
    enable_smooth_stop: true
    error_deriv_lpf_cutoff_hz: 5.0
    front_overhang: 1.0
    input_delay: 0.24
    kd: 0.0
    keep_steer_control_until_converged: false
    ki: 0.1
    kp: 1.0
    lateral_controller_mode: mpc_follower
    left_overhang: 0.128
    longitudinal_controller_mode: pid
    lpf_pitch_gain: 0.95
    lpf_vel_error_gain: 0.9
    max_acc: 3.0
    max_d_effort: 0.0
    max_i_effort: 0.3
    max_jerk: 2.0
    max_out: 1.0
    max_p_effort: 1.0
    max_pitch_rad: 0.1
    max_steer_angle: 0.0

File: control/trajectory_follower/src/mpc_lateral_controller.cpp (hunk 1 above)
Comment (takayuki5168) [reply control/trajectory_follower/src/mpc_lateral_controller.cpp]: (Sorry, this comment was pending for a long time :bow: )

Comment (TakaHoribe) [reply control/trajectory_follower/src/mpc_lateral_controller.cpp]: ok

**Pull request merged**